fhd_core/setup_metadata/fhd_struct_init_layout.pro

```diff
@@ -0,0 +1,268 @@
+  if n_elements(antenna_coords) EQ 0 then begin
+    antenna_coords=fltarr(3, n_antenna)
+  endif else begin
+    sz = size(antenna_coords)
+    if sz[0] ne 2 or sz[1] ne 3 or sz[2] ne n_antenna then message, 'antenna_coords must have shape (3, n_antenna)'
+  endelse
+  
+  ;; mount_type codes (from AIPS memo 117): 0 for alt-azimuth, 1 for equatorial, 2 for orbiting, 3 for X-Y,
+  ;; 4 for right-handed Naismith, and 5 for left-handed Naismith are defined. Aperture arrays, which are
+  ;; steered electronically rather than mechanically, are assigned code 6.
+  if n_elements(mount_type) EQ 0 then begin
+    mount_type=0
+  endif else begin
+    if n_elements(mount_type) gt 1 then begin
+      if n_elements(mount_type) ne n_antenna then message, 'mount_type must be a single integer or an array the length of n_antenna'
+      if min(mount_type) eq max(mount_type) then mount_type = mount_type[0]
+    endif
+  endelse
+  ;; from AIPS memo 117:
+  ;; The axis_offset is the position of the antenna phase reference point in the Yoke,
+  ;; relative to the antenna pedestal reference point. This is an antenna characteristic
+  ;; that should be unchanged when the antenna is moved to a new station. The X component
+  ;; of the offset is horizontal along the elevation axis and has no effect on
+  ;; interferometer phase. The Z component is vertical and approximately the nominal height
+  ;; of the elevation axis above ground for the antenna’s mount. Small variations from the
+  ;; nominal value have the same phase effect as the Z component of position, so they can
+  ;; be ignored. The value of the STAXOF column gives the value of the Y component of the
+  ;; axis offset. That component is horizontal and perpendicular to the elevation axis.
+  ;; It produces an elevation-dependent interferometer phase term and, thus, has to be
+  ;; accurately calibrated.
+  if n_elements(axis_offset) EQ 0 then begin
+    axis_offset=0
+  endif else begin
+    if n_elements(axis_offset) gt 1 then begin
+      if n_elements(axis_offset) ne n_antenna then message, 'axis_offset must be a single integer or an array the length of n_antenna'
+      if min(axis_offset) eq max(axis_offset) then axis_offset = axis_offset[0]
+    endif
+  endelse
+  
+  ;; pola and polb are the feed polarization of feed A & B respectively. Options include:
+  ;; A,B: 'X', 'Y'
+  ;; A,B: 'R', 'L'
+  ;; maybe others if feeds are not orthogonal (?)
+  if n_elements(pola) EQ 0 then begin
+    pola='X'
+  endif else begin
+    if n_elements(pola) gt 1 then begin
+      if n_elements(pola) ne n_antenna then message, 'pola must be a single integer or an array the length of n_antenna'
+      wh_diff = where(pola ne pola[0], count_diff)
+      if count_diff eq 0 then pola = pola[0]
+    endif
+  endelse
+  ;; pola/polb_orientation are the orientations given in degrees (unclear where this is measured from in AIPS memo 117)
+  if n_elements(pola_orientation) EQ 0 then begin
+    pola_orientation=0
```

Comment:
You probably want this to always be a `float`, so I recommend `pola_orientation=0.`